kernel/examples/kernel-examples/src/main/java/io/delta/kernel/examples/BaseTableWriter.java

```diff
@@ -0,0 +1,268 @@
+        return new FilteredColumnarBatch(
+                batch, // data
+                // Optional selection vector. If want to write only a subset of rows from the batch.
+                Optional.empty());
+    }
+
+
+    //////////////////////// Helper methods to create ColumnVectors ////////////////////////
+    // These are sample vectors which can be created as wrappers as engine specific       //
+    // vector types.                                                                      //
+    ////////////////////////////////////////////////////////////////////////////////////////
+    static ColumnVector intVector(List<Integer> data) {
+        return new ColumnVector() {
+            @Override
+            public DataType getDataType() {
+                return IntegerType.INTEGER;
+            }
+
+            @Override
+            public int getSize() {
+                return data.size();
+            }
+
+            @Override
+            public void close() {
+            }
+
+            @Override
+            public boolean isNullAt(int rowId) {
+                return false;
+            }
+
+            @Override
+            public int getInt(int rowId) {
+                return data.get(rowId);
+            }
+        };
+    }
+
+    static ColumnVector doubleVector(List<Double> data) {
+        return new ColumnVector() {
+            @Override
+            public DataType getDataType() {
+                return DoubleType.DOUBLE;
+            }
+
+            @Override
+            public int getSize() {
+                return data.size();
+            }
+
+            @Override
+            public void close() {
+            }
+
+            @Override
+            public boolean isNullAt(int rowId) {
+                return data.get(rowId) == null;
+            }
+
+            @Override
+            public double getDouble(int rowId) {
+                return data.get(rowId);
+            }
+        };
+    }
+
+    static ColumnVector stringVector(List<String> data) {
+        return new ColumnVector() {
+            @Override
+            public DataType getDataType() {
+                return StringType.STRING;
+            }
+
+            @Override
+            public int getSize() {
+                return data.size();
+            }
+
+            @Override
+            public void close() {
+            }
+
+            @Override
+            public boolean isNullAt(int rowId) {
+                return data.get(rowId) == null;
+            }
+
+            @Override
+            public String getString(int rowId) {
+                return data.get(rowId);
+            }
+        };
+    }
+
+    static ColumnVector stringSingleValueVector(String value, int size) {
+        return new ColumnVector() {
+            @Override
+            public DataType getDataType() {
+                return IntegerType.INTEGER;
```

Comment:
shouldn't this be STRING ?

Reply:
Yep it should. Given this is used for partition columns, which are always removed before write. That is why we didn't had any issues.